cristin-commons/src/main/java/no/unit/nva/cristin/model/CristinQuery.java

```diff
@@ -96,11 +105,50 @@ public URI toNvaFacetURI() {
      * @return an URI to NVA (default) Projects with parameters.
      */
     public URI toURI() {
-        return
-            UriWrapper.fromUri(CRISTIN_API_URL)
-                .addChild(getCristinPath())
-                .addQueryParameters(toParameters())
-                .getUri();
+        var uri = UriWrapper.fromUri(CRISTIN_API_URL)
+                      .addChild(getCristinPath());
+
+        uri = appendParameters(toParameters(), uri);
+
+        return uri.getUri();
+    }
+
+    private UriWrapper appendParameters(Map<String, String> parameters, UriWrapper uri) {
+        var multipleKeyMap = createMultipleKeyMap(parameters);
+        var appendedUri = uri;
+
+        for (NameValuePair pair : multipleKeyMap) {
+            appendedUri = addQueryParameter(appendedUri, pair);
+        }
+
+        return appendedUri;
+    }
+
+    private List<BasicNameValuePair> createMultipleKeyMap(Map<String, String> parameters) {
+        var multipleKeyMap = new ArrayList<BasicNameValuePair>();
+
+        parameters.forEach((key, values) -> {
+            if (nonNull(values)) {
+                if (multiValuedCristinParams().contains(key)) {
+                    var value = values.split(COMMA_DELIMITER_REGEX);
+                    Arrays.stream(value)
+                        .forEach(currentValue -> multipleKeyMap.add(new BasicNameValuePair(key, currentValue)));
+                } else {
+                    multipleKeyMap.add(new BasicNameValuePair(key, values));
+                }
+            }
+        });
+
+        return multipleKeyMap;
+    }
+
+    private UriWrapper addQueryParameter(UriWrapper uri, NameValuePair pair) {
+        return UriWrapper.fromUri(uri.toString())
+                   .addQueryParameter(pair.getName(), pair.getValue());
+    }
```

Comment:
I have merged nva-commons, if you bump version to 1.40.7 I think you can remove `addQueryParameter` this method.

Reply:
Will do, either in this PR or new one